docs/config-and-instances.md

```diff
@@ -0,0 +1,340 @@
+```yaml
+Kind: Config
+Metadata:
+  # If a config for a started instance should be created
+  instanceConfig: False
+  # If instanceConfig: Automatically save changes to the instance to the
+  # associated Config
+  autoSave: False
+Spec:
+  Template:
+    Metadata:
+    Spec:
+      Domain:
+        Devices:
+          …
+```
+
+The metadata `instanceConfig` and `autoSave` will be described further below.
+
+**Instance**
+
+The instance object `VM` is equal to the already present entity in the core
+KubeVirt runtime.
+
+```yaml
+Kind: VM
+Metadata:
+  ownerReferences: [{$config}]
+Spec:
+  Domain:
+    Devices:
+      …
+State:
+  Domain:
+    …
+```
+
+
+## Flow
+
+The general concept is that the `Config` contains a static VM configuration.
+Static refers to the fact, that no runtime information is kept in such objects.
+
+The runtime informations of a VM are kept in the `VM` object, i.e. it's state
+or computed values like bus addresses.
+
+## Creation
+To create a VM definition, the user needs to `POST` a `Config`.
+
+## Starting a VM, the `/start` sub-resource
+To create an instance from a `Config` a user has to `GET` the `start`
+sub-resource of a `Config` instance.
+
+## Stopping a VM, the `/stop` sub-resource
+To stop a VM, the user can `GET` the `stop` sub-resource of the associated
+`Config` object.
+
+**Note:** The semantics of a `DELETE` on the `VM` object of a `Config` still
+need to be defined.
+
+## Changing a `Config`
+Changes to a `Config` instance can be performed through the usual `PATH` calls.
+Chaging a `Config` does not lead to an automatic propagation of the change to
+the instance.
+However, in future there could be a specific metadata or action to trigger
+such a propagation automatically.
+
+
+### Example Flow
+
+Let's start with a simple example to get a better feeling for the realtionship
+between the objects.
+
+```
+$c1        $i1        $i2
+POST
+/start     POST
+/stop      DELETE
+PATCH
+/start                POST
+/stop                 DELETE
+```
+
+
+1. `POST` $c1 config
+2. `GET` $c1/start
+   → creates instance `POST` $i1
+3. `GET` $c1/stop
+   → `DELETE` $i1
+4. `PATCH` $c3
+5. `GET` $c1/start
+   → creates instance `POST` $i2
+6. `GET` $c1/stop
+   → deletes instance `DELETE` $i2
+
+In this example, a the user defines a VM, then starts it and an instance is
+getting created. Once the user stops it, the instance is getting removed.
+After changing the `Config` (in step 4), he starts and stops the VM again.
+
+This is pretty straight forward.
+
+
+### Example Flow with persistent VM
+
+In the previous example, the VM is always regenerated from the high-level
+`Config` object. This has the implication that the `VM` config can change
+between two runs. This might not be desireable in some scenarios.
+
+To solve this problem, the user can signal the system, to create a `Config`
+'snapshot' of a `VM` object, whenever it is getting created.
+The flag `instanceConfig` can be set on a `Config`, to tell the system
```

Comment:
@davidvossel not relevant anymore, but yes, we would have gotten multiple configs - the reason for this is that libvirt or kubevirt can change between the previously auto-generated config and the new start,m whcih can lead to a new expansion - which can eventually break your vm - thus in that case you could roll back to the previous config.

However, I've taken this out of the proposal. We can discuss it separately.